packages/fxa-settings/src/components/ConnectedServices/index.tsx

```diff
@@ -6,40 +6,88 @@ import React, { useState } from 'react';
 import { AlertBar } from '../AlertBar';
 import { ButtonIconReload } from '../ButtonIcon';
 import { useAlertBar } from 'fxa-settings/src/lib/hooks';
-import { useAccount, useLazyAccount } from '../../models';
+import { AttachedClient, useAccount, useLazyAccount } from '../../models';
+import { Location } from '../../models/Account';
 import { LinkExternal } from 'fxa-react/components/LinkExternal';
+import groupBy from 'lodash.groupby';
 
 const UTM_PARAMS =
   '?utm_source=accounts.firefox.com&utm_medium=referral&utm_campaign=fxa-devices';
 const DEVICES_SUPPORT_URL =
   'https://support.mozilla.org/kb/fxa-managing-devices' + UTM_PARAMS;
 
-export const Service = () => {
+function sortAndFilterConnectedClients(attachedClients: Array<AttachedClient>) {
+  const groupedByName = groupBy(attachedClients, 'name');
+
+  // get a unique (by name) list and sort by time last accessed
+  const sortedAndUniqueClients = Object.keys(groupedByName)
+    .map((key) => {
+      return groupedByName[key].sort(
+        (a: AttachedClient, b: AttachedClient) =>
+          a.lastAccessTime - b.lastAccessTime
+      )[0];
+    })
+    .sort((a, b) => a.lastAccessTime - b.lastAccessTime);
+
+  // move currently active client to the top
+  sortedAndUniqueClients.forEach((client, i) => {
+    if (client.isCurrentSession) {
+      sortedAndUniqueClients.splice(i, 1);
+      sortedAndUniqueClients.unshift(client);
+    }
+  });
+
+  return sortedAndUniqueClients;
```

Comment:
Since the OAuth clients will need to link-out to Monitor, Pocket etc., are you planning on say, maybe making a map of `'name': 'www.url-to-wherever.com'` and pulling that in when working on #5095? (We may want to bump that issue up to a 2)

Reply:
That sounds like a good option. bumped to a 2 and added your comment to the jira task